### ISSUE TITLE
`exporter` should add processes to image metadata

### ISSUE THREAD

**Issue by ekcasey** [buildpack/lifecycle#206]
### Motivation
We should add process types to the image metadata so that platforms can use that metadata to help users discover non-default processes that are available to run, and debug launch issues.

#### Happy Path
**Given** The `builder` has added the following process to the `/layers/config/metadata.toml` file
```
[[processes]]
  type = "web"
  command = "/path/to/executable some args"
  direct = false

[[processes]]
  type = "worker"
  command = "another command"
  direct = true
```
**When** `exporter` generates an image
**Then** The processes should be added to the `io.buildpacks.build.metadata` label
```json
{
  "processes": [
    {"type": "web", "command": "/path/to/executable some args", "direct": false},
    {"type": "worker", "command": "another command", "direct": true},
  ],
 ...}
```

#### Missing processes
**Given** `/layers/config/metadata.toml` has no processes
**When** `exporter` generates an image
**Then** There should be an empty process array in the `io.buildpacks.build.metadata` label
```json
{"processes": [],...}
```

**Comment by ekcasey** [comment]
Fixed by PR #208

**Comment by Zanadar** [comment]
Accepted:
```
± zm |master ✓| → docker inspect sample-app | jq ' .[0].Config.Labels| ."io.buildpacks.build.metadata"' | jq -r | jq
{
  "processes": [
    {
      "type": "web",
      "command": "java -jar target/sample-0.0.1-SNAPSHOT.jar",
      "args": null,
      "direct": false
    }
  ],
  "buildpacks": [
    {
      "id": "io.buildpacks.samples.java-maven",
      "version": "0.0.1"
    }
  ],
  "bom": [
    {
      "name": "jdk",
      "version": "",
      "metadata": null,
      "buildpack": {
        "id": "io.buildpacks.samples.java-maven",
        "version": "0.0.1"
      }
    }
  ],
  "launcher": {
    "version": "0.0.0",
    "source": {
      "git": {
        "repository": "",
        "commit": "3beb2de"
      }
    }
  }
}
```
